Wrapper/DBEngine.cpp

```diff
@@ -0,0 +1,273 @@
+﻿/*
+ * Copyright 2020 OmniSci, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "DBEngine.h"
+#include <thrift/Thrift.h>
+#include <array>
+#include <boost/filesystem.hpp>
+#include <boost/variant.hpp>
+#include <exception>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <utility>
+#include "Catalog/Catalog.h"
+#include "Import/Importer.h"
+#include "QueryEngine/ArrowResultSet.h"
+#include "QueryEngine/CompilationOptions.h"
+#include "QueryEngine/ResultSet.h"
+#include "QueryRunner/QueryRunner.h"
+#include "Shared/Logger.h"
+#include "Shared/mapdpath.h"
+#include "Shared/sqltypes.h"
+
+namespace EmbeddedDatabase {
+
+enum class ColumnType : uint32_t { eUNK, eINT, eDBL, eFLT, eSTR, eARR };
+
+/**
+ * Cursor internal implementation
+ */
+class CursorImpl : public Cursor {
+ public:
+  CursorImpl(std::shared_ptr<ResultSet> resultSet,
+             std::shared_ptr<Data_Namespace::DataMgr> dataMgr)
+      : m_resultSet(resultSet), m_DataMgr(dataMgr) {}
+
+  size_t GetColCount() { return m_resultSet->colCount(); }
+
+  size_t GetRowCount() { return m_resultSet->rowCount(); }
+
+  Row GetNextRow() {
+    auto row = m_resultSet->getNextRow(true, false);
+    if (row.empty()) {
+      std::cout << "Cursor::GetNextRow - Empty row" << std::endl;
+      return Row();
+    }
+    return Row(row);
+  }
+
+  ColumnType GetColType(uint32_t nPos) {
+    if (nPos < GetColCount()) {
+      SQLTypeInfo typeInfo = m_resultSet->getColType(nPos);
+      switch (typeInfo.get_type()) {
+        case kNUMERIC:
+        case kDECIMAL:
+        case kINT:
+        case kSMALLINT:
+        case kBIGINT:
+          return ColumnType::eINT;
+
+        case kDOUBLE:
+          return ColumnType::eDBL;
+
+        case kFLOAT:
+          return ColumnType::eFLT;
+
+        case kCHAR:
+        case kVARCHAR:
+        case kTEXT:
+          return ColumnType::eSTR;
+
+        default:
+          return ColumnType::eUNK;
+      }
+    }
+    return ColumnType::eUNK;
+  }
+
+ private:
+  std::shared_ptr<ResultSet> m_resultSet;
+  std::weak_ptr<Data_Namespace::DataMgr> m_DataMgr;
```

Comment:
Do we really need weak_ptr here? Why shared ptr is not enough?

Reply:
Any reasons naming these variables in different styles?
```suggestion
  std::weak_ptr<Data_Namespace::DataMgr> m_data_mgr;
```

Reply:
It's a part of the next PR and required by the Arrow converter. ResultSet does not provide such interface.
Naming -ok, will be fixed

Reply:
fixed